src/dwc_mapping.Rmd

```diff
@@ -348,6 +348,55 @@ write_csv(taxon, here("data", "processed", "taxon.csv"), na = "")
 ```
 
 
+```{r}
+
+```
+
+
+# Create source citation function
+
+All records in the extensions will be documented with a source, i.e. the link to the specific checklist taxon that was used for the specific piece of information. We build this function here first.
+
+```{r}
+build_source_citation <- function(extension, datasetKey_extension){
```

Comment:
I would expect the function to be `function(extension, checklists)`

Reply:
Ah right, you also want to provide the _name_ of the `datasetKey` column in the extension. On the other hand, you already assume that the other names (e.g. `input_scientificName`) are written as such, so I would just assume it here as well. For a proper function (expected to be used by others), you have to offer more freedom (let the user provide their column names), but for a quick helper function like here, I would just make a couple of assumptions (e.g. that the column is named `input_datasetKey`). So it should remain: `function(extension, checklists)`.

Reply:
Ok, I agree. In that case, I would also eliminate `checklists` from the function, as this dataframe is reused each time and is fixed. So the only argument in the function would be `extension`

Reply:
No, you should assume a function works in its own space: you have to pass all the information it needs.